### PR TITLE
gotosocial: 0.17.0 -> 0.17.3

### PR DESCRIPTION
**Pull request by CherryKitten**
## Things done
Update to https://github.com/superseriousbusiness/gotosocial/releases/tag/v0.17.1

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by jugendhacker — APPROVED** [review]

**Review by blakesmith — APPROVED** [review]

**Comment by r-vdp** [comment]
Seems that this broke the tests.

**Comment by blakesmith** [comment]
It looks like the tests are now failing to create the admin user:

```
>>> machine.execute("/nix/store/2rkfnmlznhg6xb66clavz9yl355h6brw-gotosocial-0.17.1/bin/gotosocial --config-path /nix/store/dybnlrwkxba6d75kxdwqmhppb594kfpv-config.yml admin account create --username nickname --email email@example.com --password kurtz575VPeBgjVm")
machine # [   40.630040] postgres[978]: [978] LOG:  provided user name (gotosocial) and authenticated user name (root) do not match
machine # 2024/11/04 06:15:39 error executing command: error creating dbConn: postgres ping: failed to connect to `user=gotosocial database=gotosocial`: /run/postgresql/.s.PGSQL.5432 (/run/postgresql): server error: FATAL: Peer authentication failed for user "gotosocial" (SQLSTATE 28000)
machine # [   40.634190] postgres[978]: [978] FATAL:  Peer authentication failed for user "gotosocial"
machine # [   40.634768] postgres[978]: [978] DETAIL:  Connection matched file "/nix/store/vk59q60l877xxmw82v12lwkwlclp7z6v-pg_hba.conf" line 3: "local all all              peer"
```

Not sure what changed, but it looks like the root user now doesn't have local unix socket access to the gotosocial postgres database. I attempted to run the admin user create command as the `gotosocial` user, but that has some issues as well:

```
diff --git a/nixos/tests/web-apps/gotosocial.nix b/nixos/tests/web-apps/gotosocial.nix
index f9d28c2b8b99..64233768c72d 100644
--- a/nixos/tests/web-apps/gotosocial.nix
+++ b/nixos/tests/web-apps/gotosocial.nix
@@ -22,7 +22,7 @@

     # check user registration via cli
     machine.succeed("curl -sS -f http://localhost:8081/nodeinfo/2.0 | jq '.usage.users.total' | grep -q '^0$'")
-    machine.succeed("gotosocial-admin account create --username nickname --email email@example.com --password kurtz575VPeBgjVm")
+    machine.succeed("sudo su -s /run/current-system/sw/bin/sh -c 'gotosocial-admin account create --username nickname --email email@example.com --password kurtz575VPeBgjVm' gotosocial")
     machine.succeed("curl -sS -f http://localhost:8081/nodeinfo/2.0 | jq '.usage.users.total' | grep -q '^1$'")
   '';
 }
```

A few options, if anyone has any thoughts / ideas:
1. Figure out how to grant `root` user permission to the local postgres database.
2. Figure out how to successfully run the admin user create commands as the `gotosocial` user.
3. Switch the tests to use sqlite instead of booting a postgres database.
4. Something else?

**Comment by CherryKitten** [comment]
> Seems that this broke the tests.

damn, sorry for missing this, I ran the tests locally and somehow did not notice it seems

**Comment by jugendhacker** [comment]
@blakesmith I would go with option 3 as SQLite is the preferred database by upstream anyways

**Comment by CherryKitten** [comment]
I just pushed another commit switching the nixos tests to sqlite, but I can't check if they pass right now, can someone else please check?

File: nixos/tests/web-apps/gotosocial.nix [hunk 1]
```diff
@@ -7,17 +7,15 @@
...
       settings = {
         host = "localhost:8081";
         port = 8081;
+        db-type = "sqlite";
       };
     };
   };
 
   testScript = ''
-    machine.wait_for_unit("gotosocial.service")
```
Comment (jugendhacker) [inline nixos/tests/web-apps/gotosocial.nix:19]: I think it should still wait for the gotosocial service to start, or?

**Comment by blakesmith** [comment]
Hrm, I'm still getting test failures. I might be wrong about the postgresql problem.

I wonder if this is an upstream issue that was released in 0.17.0. If you checkout what's in nixpkgs-unstable at the moment, we're getting the same test failures:

```
$ git checkout origin/nipkgs-unstable
$ nix run '.#nixosTests.gotosocial.driver'
```

Gives us the same failure. Did user creation get moved to something async, or perhaps there's some other bug?

```
machine # Connection to localhost (127.0.0.1) 8081 port [tcp/sunproxyadmin] succeeded!
(finished: waiting for TCP port 8081 on localhost, in 10.32 seconds)
machine: must succeed: curl -sS -f http://localhost:8081/nodeinfo/2.0 | jq '.usage.users.total' | grep -q '^0$'
machine # [   17.829100] gotosocial[903]: timestamp="04/11/2024 14:08:34.895" func=server.init.func1.Logger.13.1 level=INFO latency="2.536076ms" userAgent=curl/8.9.1 method=GET statusCode=200 path=/nodeinfo/2.0 clientIP=127.0.0.1 requestID=9j583xwj040018xf5jbg msg="OK: wrote 222B"
(finished: must succeed: curl -sS -f http://localhost:8081/nodeinfo/2.0 | jq '.usage.users.total' | grep -q '^0$', in 0.03 seconds)
machine: must succeed: gotosocial-admin account create --username nickname --email email@example.com --password kurtz575VPeBgjVm
machine # [   17.855204] systemd[1]: Starting /nix/store/fzpv16gvayhl3cbflbqjw385i4yxnzs6-gotosocial-0.17.0/bin/gotosocial --config-path /nix/store/l8zd4y9gylp24v4gddahmvvszqy4ag5c-config.yml admin account create --username nickname --email email@example.com --password kurtz575VPeBgjVm...
machine # [   17.874233] systemd[1]: Started /nix/store/fzpv16gvayhl3cbflbqjw385i4yxnzs6-gotosocial-0.17.0/bin/gotosocial --config-path /nix/store/l8zd4y9gylp24v4gddahmvvszqy4ag5c-config.yml admin account create --username nickname --email email@example.com --password kurtz575VPeBgjVm.
machine # [   18.069841] systemd[1]: gotosocial-admin.service: Deactivated successfully.
(finished: must succeed: gotosocial-admin account create --username nickname --email email@example.com --password kurtz575VPeBgjVm, in 0.24 seconds)
machine: must succeed: curl -sS -f http://localhost:8081/nodeinfo/2.0 | jq '.usage.users.total' | grep -q '^1$'
machine # [   18.095610] gotosocial[903]: timestamp="04/11/2024 14:08:35.162" func=server.init.func1.Logger.13.1 level=INFO latency="64.534µs" userAgent=curl/8.9.1 method=GET statusCode=200 path=/nodeinfo/2.0 clientIP=127.0.0.1 requestID=ba5r3xwj04000bwn25s0 msg="OK: wrote 222B"
machine: output:
cleanup
kill machine (pid 2359028)
qemu-kvm: terminating on signal 15 from pid 2349838 (/nix/store/wfbjq35kxs6x83c3ncpfxdyl5gbhdx4h-python3-3.12.6/bin/python3.12)
kill vlan (pid 2359026)
(finished: cleanup, in 0.01 seconds)
Traceback (most recent call last):
  File "/nix/store/59ffvnvk82anwk2m550jq6hkxrcad0l6-nixos-test-driver-1.1/bin/.nixos-test-driver-wrapped", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/59ffvnvk82anwk2m550jq6hkxrcad0l6-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/__init__.py", line 146, in main
    driver.run_tests()
  File "/nix/store/59ffvnvk82anwk2m550jq6hkxrcad0l6-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/driver.py", line 175, in run_tests
    self.test_script()
  File "/nix/store/59ffvnvk82anwk2m550jq6hkxrcad0l6-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/driver.py", line 167, in test_script
    exec(self.tests, symbols, None)
  File "<string>", line 8, in <module>
  File "/nix/store/59ffvnvk82anwk2m550jq6hkxrcad0l6-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/machine.py", line 611, in succeed
    raise Exception(f"command `{command}` failed (exit code {status})")
Exception: command `curl -sS -f http://localhost:8081/nodeinfo/2.0 | jq '.usage.users.total' | grep -q '^1$'` failed (exit code 1)
```

If you run the tests with `nix run '.#nixosTests.gotosocial.driverInteractive'`, you can see that the admin user create succeeds, but the user count in the curl output never gets incremented:

```
$ nix run '.#nixosTests.gotosocial.driverInteractive
>>> start_all()
>>> run_tests()
>>> machine.execute("curl -sS -f http://localhost:8081/nodeinfo/2.0 | jq -c '.'")
machine # [   90.753716] gotosocial[868]: timestamp="04/11/2024 14:14:39.775" func=server.init.func1.Logger.13.1 level=INFO latency="64.534µs" userAgent=curl/8.9.1 method=GET statusCode=200 path=/nodeinfo/2.0 clientIP=127.0.0.1 requestID=krdrfxwj04000sg75seg msg="OK: wrote 222B"
(0, '{"version":"2.0","software":{"name":"gotosocial","version":"0.17.0"},"protocols":["activitypub"],"services":{"inbound":[],"outbound":[]},"openRegistrations":false,"usage":{"users":{"total":0},"localPosts":0},"metadata":{}}\n')
```

**Comment by CherryKitten** [comment]
Upstream just released https://github.com/superseriousbusiness/gotosocial/releases/tag/v0.17.2 so I updated the PR accordingly

File: pkgs/servers/gotosocial/default.nix [hunk 2]
```diff
@@ -9,11 +9,11 @@ let
   owner = "superseriousbusiness";
   repo = "gotosocial";
 
-  version = "0.17.0";
+  version = "0.17.1";
```
Comment (blakesmith) [inline pkgs/servers/gotosocial/default.nix:12]: Did you mean for this to be 0.17.2?

**Comment by blakesmith** [comment]
Logged a separate issue for the failing NixOS tests: https://github.com/NixOS/nixpkgs/issues/353896

File: pkgs/servers/gotosocial/default.nix (hunk 2 above)
Comment (CherryKitten) [reply pkgs/servers/gotosocial/default.nix:12]: what.. did i forget to git add before amending? I'll check and fix it in the morning..

Comment (CherryKitten) [reply pkgs/servers/gotosocial/default.nix]: it's fixed now, sorry about the mess ^^

**Comment by CherryKitten** [comment]
Upstream released another bugfix release (https://github.com/superseriousbusiness/gotosocial/releases/tag/v0.17.3) so I'm updating the PR for hopefully one last time

**Comment by blakesmith** [comment]
@CherryKitten Ha! They're moving fast! Thanks for being on top of it. :grin: 

Would you also mind weighing in on https://github.com/NixOS/nixpkgs/pull/353901? Will be good to get the tests passing again to speed up these version bumps.

**Review by blakesmith — APPROVED** [review]

**Review by jugendhacker — APPROVED** [review]

**Review by blakesmith — APPROVED** [review]

**Comment by blakesmith** [comment]
@CherryKitten Looks like `gotosocial` just got picked up, and migrated to the new pkgs/by-name standard: https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/go/gotosocial/package.nix in this commit here: https://github.com/NixOS/nixpkgs/commit/571c71e6f73af34a229414f51585738894211408.

Sorry for more hassle, but would you mind updating your PR to the new package location?

**Comment by CherryKitten** [comment]
> @CherryKitten Looks like `gotosocial` just got picked up, and migrated to the new pkgs/by-name standard: https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/go/gotosocial/package.nix in this commit here: [571c71e](https://github.com/NixOS/nixpkgs/commit/571c71e6f73af34a229414f51585738894211408).
> 
> Sorry for more hassle, but would you mind updating your PR to the new package location?

done, sorry I didn't get to it during the weekend

Let's hope that's the last change to this PR before we can get it merged lol

**Review by blakesmith — APPROVED** [review]

**Review by blakesmith — APPROVED** [review]

**Comment by emilazy** [comment]
@ofborg test gotosocial

**Review by leona-ya — APPROVED** [review]
Tests are now successful with the fixed nixos tests (when rebased on master).

**Comment by emilazy** [comment]
Probably worth backporting to 24.11?

**Pull request merged**